pkg/injection/namespace/bootstrapperconfig/endpoints.go

```diff
@@ -0,0 +1,66 @@
+package bootstrapperconfig
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/Dynatrace/dynatrace-bootstrapper/pkg/configure/enrichment/endpoint"
+	"github.com/Dynatrace/dynatrace-operator/pkg/api/v1beta3/dynakube"
+	dtclient "github.com/Dynatrace/dynatrace-operator/pkg/clients/dynatrace"
+	dtingestendpoint "github.com/Dynatrace/dynatrace-operator/pkg/injection/namespace/ingestendpoint"
+	k8ssecret "github.com/Dynatrace/dynatrace-operator/pkg/util/kubeobjects/secret"
+	"github.com/pkg/errors"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func (s *SecretGenerator) prepareEndpoints(ctx context.Context, dk *dynakube.DynaKube) (map[string][]byte, error) {
+	fields, err := s.prepareFieldsForEndpoints(ctx, dk)
+	if err != nil {
+		return nil, errors.WithStack(err)
+	}
+
+	endpointPropertiesBuilder := strings.Builder{}
+
+	if dk.MetadataEnrichmentEnabled() {
+		if _, err := endpointPropertiesBuilder.WriteString(fmt.Sprintf("%s=%s\n", dtingestendpoint.MetricsUrlSecretField, fields[dtingestendpoint.MetricsUrlSecretField])); err != nil {
+			return nil, errors.WithStack(err)
+		}
+
+		if _, err := endpointPropertiesBuilder.WriteString(fmt.Sprintf("%s=%s\n", dtingestendpoint.MetricsTokenSecretField, fields[dtingestendpoint.MetricsTokenSecretField])); err != nil {
+			return nil, errors.WithStack(err)
+		}
+	}
+
+	data := map[string][]byte{
+		endpoint.InputFileName: bytes.NewBufferString(endpointPropertiesBuilder.String()).Bytes(),
+	}
+
+	return data, nil
```

Comment:
you only need the string data here
```suggestion
	return endpointPropertiesBuilder.String(), nil
```

then in `generate`
```go
	return map[string][]byte{
		pmc.InputFileName:        pmcSecret,
		ca.TrustedCertsInputFile: trustedCAs,
		ca.AgCertsInputFile:      agCerts,
		curl.InputFileName:       []byte(initialConnectRetryMs),
		endpoint.InputFileName:   []byte(endpointProperties),  // <= this
	}, nil
```

Reply:
done https://github.com/Dynatrace/dynatrace-operator/pull/4515/commits/21af4e8e16b34ef966cb4a5d6a4a765958b0fc97

and added some unit tests